File: C#/LabWork#5/Digits.cs
using System;


namespace LabWork_5
{
  public class Digits : Row
  {
    public Digits(string val) : base(val) { }

    public override void Reduce()
    {
      string temp = "";
      for (int i = 0; i < value.Length; i++)
      {
        if (i % 2 != 0)
          temp += value[i];
      }
      value = temp;
    }
  }


}

File: C++/LabWork#5/Digits.cpp
#include "Digits.h"

Digits::Digits(std::string val) : Row(val) {}

void Digits::reduce() {
  std::string temp = "";
  for (int i = 0; i < value.size(); i++) {
    if (i % 2 != 0)
      temp += value[i];
  }
  value = temp;
}
